src/test/scala/performanceanalysis/logreceiver/LogReceiverActorSpec.scala

```diff
@@ -12,21 +12,56 @@ class LogReceiverActorSpec(testSystem: ActorSystem) extends ActorSpecBase(testSy
   def this() = this(ActorSystem("LogReceiverActorSpec"))
 
   "LogReceiverActor" must {
-    val logReceiverActor = system.actorOf(LogReceiverActor.props)
+    val compId = "newComponent1"
+    val logParserProbe = TestProbe("logParserProbe")
+
     "handle incoming RegisterNewLogParser messages by adding the new actor to its context" in {
+      val logReceiverActor = TestActorRef(LogReceiverActor.props)
       val testProbe = TestProbe("testProbe")
-      val logParserProbe = TestProbe("logParserProbe")
-      val componentName = "newComponent1"
-      val logLine = "test log line"
+      val logLines = "log line1\nlog line2"
 
       // Verify that LogParserNotFound is returned when posting a log on an unknown component
-      testProbe.send(logReceiverActor, SubmitLog(componentName, "test log line"))
-      testProbe.expectMsgPF() { case LogParserNotFound(`componentName`) => true }
+      testProbe.send(logReceiverActor, SubmitLogs(compId, logLines))
+      testProbe.expectMsgPF() { case LogParserNotFound(`compId`) => true }
 
       // Register a new actor and verify that logs are now accepted for processing
-      testProbe.send(logReceiverActor, RegisterNewLogParser(componentName, logParserProbe.ref))
-      testProbe.send(logReceiverActor, SubmitLog(componentName, logLine))
-      testProbe.expectMsgPF() { case LogSubmitted => true }
+      testProbe.send(logReceiverActor, RegisterNewLogParser(compId, logParserProbe.ref))
+      EventFilter.debug(start = s"New LogParser created with $compId", occurrences = 1) intercept {
+        logReceiverActor ! RegisterNewLogParser(compId, logParserProbe.ref)
+      }
+
+      testProbe.send(logReceiverActor, SubmitLogs(compId, logLines))
+
+      logParserProbe.expectMsg(SubmitLog(compId, "log line1"))
+      logParserProbe.expectMsg(SubmitLog(compId, "log line2"))
+
+      testProbe.expectMsg(LogsSubmitted)
+    }
+  }
+
+
+  "splitIntoLines" must {
+    import LogReceiverActor.splitIntoLines
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Put imports in the beginning of the package](https://www.codacy.com/app/scala-academy/performance-analysis/file/2633088833/issues/source?bid=3220915&fileBranchId=3313889#l44)